tests/syntax/test_specifiers.py

```diff
@@ -47,181 +52,501 @@ def test_lazy_value_in_requirement():
+    )
+    for i in range(20):
+        scene = sampleScene(scenario, maxIterations=10)
+        assert scene.egoObject.position.distanceTo(Vector(300, 200, 500)) <= 2
+
 def test_visible_from_oriented_point():
     scenario = compileScenic(
-        'op = OrientedPoint at 100 @ 200, facing 45 deg,\n'
+        'op = new OrientedPoint at 100 @ 200, facing 45 deg,\n'
         '                   with visibleDistance 5, with viewAngle 90 deg\n'
-        'ego = Object visible from op'
+        'ego = new Object visible from op'
     )
     base = Vector(100, 200)
-    for i in range(30):
-        scene = sampleScene(scenario, maxIterations=1)
+    for i in range(20):
+        scene = sampleScene(scenario, maxIterations=10)
         pos = scene.egoObject.position
         assert pos.distanceTo(base) <= 5
         assert pos.x <= base.x
         assert pos.y >= base.y
 
+@pytest.mark.slow
+def test_can_see_point_occlusion_enabled():
+    with pytest.raises(RejectionException):
+        sampleSceneFrom("""
+            workspace_region = RectangularRegion(0 @ 0, 0, 40, 40)
+            workspace = Workspace(workspace_region)
+
+            ego = new Object with visibleDistance 30,
+                at (0,0,1),
+                with width 5,
+                with length 5,
+                with height 5,
+                with pitch 45 deg,
+                with viewAngles (340 deg, 60 deg),
+                with rayDensity 5
+
+            seeing_pt = new Point at (0,10,5),
+                with name "seeingPoint",
+                visible from ego
+
+            new Object at (0,5,4),
+                with width 10,
+                with length 0.5,
+                with height 6,
+                with name "wall",
+        """, maxIterations=1)
+
+@pytest.mark.slow
+def test_point_visible_from_object():
+    sampleSceneFrom("""
+        workspace_region = RectangularRegion(0 @ 0, 0, 40, 40)
+        workspace = Workspace(workspace_region)
+
+        ego = new Object with visibleDistance 30,
+            at (0,0,1),
+            with width 5,
+            with length 5,
+            with height 5,
+            with pitch 45 deg,
+            with viewAngles (340 deg, 60 deg),
+            with rayDensity 5
+
+        seeing_pt = new Point at (0,10,5),
+            with name "seeingPoint",
+            visible from ego
+
+        new Object at (0,5,4),
+            with width 10,
+            with length 0.5,
+            with height 6,
+            with name "wall",
+            with occluding False
+    """, maxIterations=1)
+
 def test_not_visible():
     scenario = compileScenic("""
         workspace = Workspace(RectangularRegion(100@205, 0, 20, 12))
-        ego = Object at 100 @ 200, facing -45 deg,
+        ego = new Object at 100 @ 200, facing -45 deg,
                      with visibleDistance 10, with viewAngle 90 deg
-        ego = Object not visible
+        ego = new Object in workspace, not visible
```

Comment:
This test was supposed to check that `not visible` works by itself to specify `position` (i.e., the `in workspace` should not be necessary); if you want to test the new behavior of also adding a visibility requirement, that should be a separate test.

Reply:
Split into two tests, one for 3D with a box region one for 2D with the old rectangular region and 2D compatibility mode on.